analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/TestHelper.cs

```diff
@@ -255,27 +253,32 @@ public static string CreateAnalysisConfig(TestContext context, IEnumerable<strin
                 </AnalysisConfig>
                 """);
 
+        public static string CreateSonarProjectConfigWithFilesToAnalyze(TestContext context, params string[] filesToAnalyze)
+        {
+            var filesToAnalyzePath = TestPath(context, "FilesToAnalyze.txt");
+            File.WriteAllLines(filesToAnalyzePath, filesToAnalyze);
+            return CreateSonarProjectConfig(context, "FilesToAnalyzePath", filesToAnalyzePath, true);
+        }
+
         public static string CreateSonarProjectConfig(TestContext context, IEnumerable<string> unchangedFiles)
         {
             var analysisConfigPath = CreateAnalysisConfig(context, unchangedFiles);
-            return CreateSonarProjectConfig(Path.GetDirectoryName(analysisConfigPath), "AnalysisConfigPath", analysisConfigPath, true);
+            return CreateSonarProjectConfig(context, "NotImporant", null, true, analysisConfigPath);
```

Comment:
It can also be a one liner:
```
        public static string CreateSonarProjectConfig(TestContext context, IEnumerable<string> unchangedFiles) =>
            CreateSonarProjectConfig(context, "NotImportant", null, true, CreateAnalysisConfig(context, unchangedFiles));
```